crates/ruff_linter/src/settings/flags.rs

```diff
@@ -1,8 +1,24 @@
 #[derive(Debug, Copy, Clone, Hash, is_macro::Is)]
 pub enum FixMode {
-    Generate,
+    Generate(SuggestedFixes),
+    Apply(SuggestedFixes),
+    Diff(SuggestedFixes),
+}
```

Comment:
it feels like this should be a struct with two enums, but not critical to merging